### ISSUE TITLE
[13] Use init systems to manage bots and botnet

### ISSUE THREAD

**Issue by sebix** [certtools/intelmq#693]

**Comment by ondj** [comment]
I think intelmqctl should be startup system independent because, for example, I use intelmqctl for starting and stopping bots running under supervisord in Docker container.

**Comment by sebix** [comment]
Sure, providing a configuration for supervisord is no problem. But as default, I would not use supervisor, as it's not packaged on lts distros.

**Comment by sebix** [comment]
Moving this to 1.0, see https://github.com/certtools/intelmq/pull/763#issuecomment-260991676

**Comment by sebix** [comment]
To create/update the systemd/supervisord config we could implement the functionality in intelmqctl e.g. `intelmqctl create-scripts` which will then "convert" the runtime config into systemd services/supervisord config (depending on configuration).

However, this needs to be called as root, as `/lib/systemd/system`. Or we place them under a intelmq-writeable directory and use the full path when calling `systemctl`.

**Comment by ondj** [comment]
And what about to use [systemd generator](https://www.freedesktop.org/software/systemd/man/systemd.generator.html)?

But maybe it is not suitable, because:

> Generators are run very early at boot and cannot rely on any external services. They may not talk to any other process. That includes simple things such as logging to syslog(3), or systemd itself (this means: no systemctl(1))! Non-essential file systems like /var and /home are mounted after generators have run. Generators can however rely on the most basic kernel functionality to be available, including a mounted /sys, /proc, /dev, /usr.

**Comment by wagner-certat** [comment]
Thanks for the pointer!

We definitely need `/etc` to load the runtime.

Also important:

> If you are careful, you can implement generators in shell scripts. We do recommend C code however, since generators are executed synchronously and hence delay the entire boot if they are slow.

**Comment by wagner-certat** [comment]
@certtools/intelmq-contributors 
I think we should postpone this to a later release. The current base is very stable and we postponed the 1.0 release already for too long. As introducing such a change right before beta/RC-releases means a lot of discussions, work and rash decisions, I'd rather not do this now.

I changed my opinion in this regard, this change is not *necessary* for the next stable release. Big changes in the process management implicates some months of intense discussions, implementation and testing.
